README.md

```diff
@@ -164,6 +164,13 @@ the following fields:
   implementations that restart the system as part of their normal operation
   and expect to start up with the same config after a restart.
 
+  **WARNING**: in environments where tasks runs on the host operating system,
+  this functionality can expose the worker's credentials to the task.  The
+  process creates this configuration file with read permission only for the
+  current user. As an added layer of protection, we recommend using a filename
+  located in a directory that is only accessible by the user running
+  `start-worker`. 
+
```

Comment:
Note, `generic-worker` actively takes ownership of the config file on startup, making it unreadable to task users, so this isn't a concern for generic-worker. On Windows, the go `os` package functions mostly ignore the file permission flags that are passed in. Instead, explicit ACLs typically need to be created, but this is handled by generic-worker [here](https://github.com/taskcluster/generic-worker/blob/955969c19e076c5918440f18122428acc04bc7ba/multiuser.go#L26).

Maybe we should update the comment to say that worker runner expects the worker implementation to secure the config file?

Reply:
This isn't the generic-worker config file -- it's the cache for worker-runner's state.

I didn't know that about the `os` package -- I looked into its source and it *seemed* to convert that to something useful, but maybe it doesn't work.  I can replicate the behavior of generic-worker just to be sure.

Reply:
I ended up using github.com/hectane/go-acl since its interface is nice and simple and since it uses the owner SID directly, rather than assuming Administrator is the right group.

Reply:
We used to use github.com/hectane/go-acl too in generic-worker until [bug 1594353](https://bugzilla.mozilla.org/show_bug.cgi?id=1594353). The bug was actually a generic-worker bug, but using go-acl meant that we didn't see it on Windows 7 and Windows 10. I therefore decided to change from using go-acl to using `icacls` so we'd get clearer output in case of problems, and passing in bad values would be handled correctly.

I discovered that the file permission bits of the `os` package functions weren't respected on Windows [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1439588#c38). That bug has a lot of useful context on the topic.